agr_literature_service/api/crud/reference_crud.py

```diff
@@ -864,7 +854,36 @@ def sql_query_for_missing_files(db: Session, mod_abbreviation: str, order_by: st
     if offset is not None:
         params['offset'] = str(offset)
 
-    return query.bindparams(**params)
+    rows = db.execute(query, params).mappings().fetchall()
+    ref_data = jsonable_encoder(rows)
+    if not ref_data:
+        return ref_data
+
+    reference_ids = [item['reference_id'] for item in ref_data if 'reference_id' in item]
+    reference_ids_sql_txt = ', '.join(str(item) for item in reference_ids)
```

Comment:
Can we  eliminate the intermediate reference_ids_sql_txt and directly use bind parameter as follows? 

reference_ids = [item['reference_id'] for item in ref_data if 'reference_id' in item]
reffile_query_str = """
    SELECT reference_id,
           COUNT(1) FILTER (WHERE file_class = 'main') AS maincount,
           COUNT(1) FILTER (WHERE file_class = 'supplement') AS supcount
    FROM referencefile
    WHERE reference_id = ANY(:reference_ids)
    GROUP BY reference_id
"""

rows_reffile = db.execute(text(reffile_query_str), {'reference_ids': reference_ids}).mappings().fetchall()

Reply:
Oh, cool !  I didn't know the syntax for that, that looks much better, I'll try it after stand up, thanks !

Reply:
The fix looks great. Thanks